iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/ExcessivePermissionsCheck.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.terraform.checks.gcp;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.iac.common.api.checks.SecondaryLocation;
+import org.sonar.iac.common.checks.TextUtils;
+import org.sonar.iac.terraform.api.tree.ExpressionTree;
+import org.sonar.iac.terraform.checks.AbstractNewResourceCheck;
+import org.sonar.iac.terraform.symbols.ListSymbol;
+
+@Rule(key = "S6406")
+public class ExcessivePermissionsCheck extends AbstractNewResourceCheck {
+
+  private static final String MESSAGE = "This role grants more than %d sensitive permissions. Make sure they are all required.";
+  private static final String SECONDARY_MESSAGE = "Sensitive permission";
+  public static final int DEFAULT = 5;
+
+  private static final Set<String> SENSITIVE_FEATURE_SUFFIXES = Set.of("abort", "access", "add", "allocate", "analyze", "apply", "approve", "associate", "attach", "begin",
+      "bind", "call", "cancel", "clear", "close", "compute", "connect", "create", "delete", "deploy", "destroy", "detach",
+      "disable", "drop", "enable", "evict", "exec", "import", "install", "invoke", "listVulnerabilities", "manage",
+      "migrate", "move", "mutate", "patch", "pause", "proxy", "publish", "purchase", "purge", "put", "reject", "remove",
+      "reopen", "replace", "rerun", "reset", "resize", "restart", "restore", "resume", "rollback", "rotate", "run",
+      "sample", "scan", "send", "set", "sign", "sourceCodeGet", "sourceCodeSet", "start", "stop", "suspend", "undelete",
+      "undeploy", "update", "upload", "use", "validate", "write");
+
+  private static final Set<String> SENSITIVE_FEATURE_SUFFIX_ELEMENTS = Set.of("login", "create", "delete", "set");
```

Comment:
For analogous reasons, if you choose to go for SENSITIVE_SUFFIX_PREFIXES in the above case, here this should be renamed to simply **SENSITIVE_SUFFIX_ELEMENTS**
(If you go for SENSITIVE_FEATURE_SUFFIX_PREFIXES in the above case, then here it should remain as it is)

Reply:
SENSITIVE_ACTIONS_ELEMENTS?

Reply:
(I've addressed this in the above comment)